### PR TITLE
Update taint key,value documentation

### PR DESCRIPTION
**Pull request by ajitak**
**What this PR does / why we need it**: Current taint key, value documentation is not correct, fix it.

```NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/taint.go [hunk 1]
```diff
@@ -57,8 +57,8 @@ var (
 		Update the taints on one or more nodes.
 
 		* A taint consists of a key, value, and effect. As an argument here, it is expressed as key=value:effect.
-		* The key must begin with a letter or number, and may contain letters, numbers, hyphens, dots, and underscores, up to %[1]d characters.
-		* The value must begin with a letter or number, and may contain letters, numbers, hyphens, dots, and underscores, up to %[2]d characters.
+		* The key must be %[1]d characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.
+		* The value must be %[2]d characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.
```
Comment (ravisantoshgudimetla) [inline pkg/kubectl/cmd/taint.go]: Probably, we can add `less, must be empty or begin and end with`. Just to be consistent.

**Comment by ajitak** [comment]
/test pull-kubernetes-e2e-gce-bazel
--

File: pkg/kubectl/cmd/taint.go (hunk 1 above)
Comment (ajitak) [reply pkg/kubectl/cmd/taint.go]: I have copied this text from https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set to be consistent with labels.

It looks to me that key/value cannot be empty.

**Comment by ajitak** [comment]
Ping

File: pkg/kubectl/cmd/taint.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply pkg/kubectl/cmd/taint.go]: @ajitak Apologies. I should have been clear. The comment I gave was intended for label values only. I copied from the same document that you linked. If you look at 2nd para in there, it tells:

`Valid label values must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.
Label selectors`

For example, b=' ', c=' ' are valid label key, values.

**Pull request closed without merging**